pkg/render/logstorage/esgateway/esgateway.go

```diff
@@ -180,15 +182,40 @@ func (e esGateway) esGatewayDeployment() *appsv1.Deployment {
 		}},
 	}
 
-	volumes := []corev1.Volume{
-		{
+	certVolume := corev1.Volume{
+		Name: VolumeName,
+		VolumeSource: corev1.VolumeSource{
+			Secret: &corev1.SecretVolumeSource{
+				SecretName: render.TigeraElasticsearchCertSecret,
+			},
+		},
+	}
+
+	var initContainers []corev1.Container
+	if e.installation.CertificateManagement != nil {
+		svcDNSNames := dns.GetServiceDNSNames(render.ElasticsearchServiceName, render.ElasticsearchNamespace, e.clusterDomain)
+		svcDNSNames = append(svcDNSNames, dns.GetServiceDNSNames(ServiceName, render.ElasticsearchNamespace, e.clusterDomain)...)
+
+		initContainers = append(initContainers, render.CreateCSRInitContainer(
+			e.installation.CertificateManagement,
+			e.csrImage,
+			VolumeName,
+			ServiceName,
+			corev1.TLSPrivateKeyKey,
+			corev1.TLSCertKey,
+			svcDNSNames,
+			render.ElasticsearchNamespace))
+
+		certVolume = corev1.Volume{
```

Comment:
Rather than a new volume object, it looks like you can change the volumesource.

Reply:
Good catch, updated